dm/checker/check_test.go

```diff
@@ -173,6 +173,44 @@ func TestReplicationPrivilegeChecking(t *testing.T) {
 	}, cfgs)
 }
 
+func TestRunCheckOnConfigsLimit(t *testing.T) {
+	cfgs := []*config.SubTaskConfig{
+		{
+			Mode: config.ModeAll,
+			IgnoreCheckingItems: ignoreExcept(map[string]struct{}{
+				config.DumpPrivilegeChecking:        {},
+				config.ReplicationPrivilegeChecking: {},
+			}),
+		},
+	}
+
+	// with limit = 100, there should be 2 error
+
+	mock := initMockDB(t)
+	mock.ExpectQuery("SHOW GRANTS").WillReturnRows(sqlmock.NewRows([]string{"Grants for User"}).
+		AddRow("GRANT USAGE ON *.* TO 'haha'@'%'"))
+	mock.ExpectQuery("SHOW GRANTS").WillReturnRows(sqlmock.NewRows([]string{"Grants for User"}).
+		AddRow("GRANT USAGE ON *.* TO 'haha'@'%'"))
+
+	result, err := RunCheckOnConfigs(context.Background(), cfgs, false, 100, 100)
+	require.NoError(t, err)
+	require.EqualValues(t, 2, result.Summary.Failed)
+	require.Len(t, result.Results, 2)
+
+	// test limit = 1
+
+	mock = initMockDB(t)
+	mock.ExpectQuery("SHOW GRANTS").WillReturnRows(sqlmock.NewRows([]string{"Grants for User"}).
+		AddRow("GRANT USAGE ON *.* TO 'haha'@'%'"))
+	mock.ExpectQuery("SHOW GRANTS").WillReturnRows(sqlmock.NewRows([]string{"Grants for User"}).
+		AddRow("GRANT USAGE ON *.* TO 'haha'@'%'"))
+
+	result, err = RunCheckOnConfigs(context.Background(), cfgs, false, 0, 1)
+	require.NoError(t, err)
+	require.EqualValues(t, 2, result.Summary.Failed)
+	require.Len(t, result.Results, 1)
```

Comment:
Can you test another case? multi errors contained by one result, and they are limited by `input:limit`.

Reply:
https://github.com/pingcap/tiflow/pull/11061/commits/4ba16262fbfd2d54c9612031ad2029ae8f728374